### PR TITLE
server-tests: unify XL and FS tests into common code. server_test.go contains common test code. server_xl_test.go contains XL tests specific to XL.

### PR DESCRIPTION
**Pull request by krishnasrinivas**

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -33,32 +33,36 @@ import (
...
 	secretKey  string
 }
 
-// Initializing the test suite.
-var _ = Suite(&TestSuiteFS{})
+// Init and run test on FS backend.
+var _ = Suite(&TestSuiteCommon{serverType: "FS"})
+
+// Init and run test on XL backend.
+var _ = Suite(&TestSuiteCommon{serverType: "XL"})
```
Comment (hackintoshrao) [inline server_test.go:49]: This guarantees running of the tests twice right ? one for "XL" and one for "FS" ? Just checking, haven't used it like that before.

**Comment by hackintoshrao** [comment]
This is a  much needed change, cleaner/simple/extensible. Have just one comment on the code, if thats good then LGTM 👍 . Thanks !

File: server_test.go (hunk 1 above)
Comment (harshavardhana) [inline server_test.go:49]: > This guarantees running of the tests twice right ? one for "XL" and one for "FS" ? Just checking, haven't used it like that before.

One has to validate how many tests were run using 'go test'

Comment (krishnasrinivas) [inline server_test.go:49]: > This guarantees running of the tests twice right ? one for "XL" and one for "FS" ? Just checking, haven't used it like that before.

Verified once again, yes works as expected.

**Pull request merged**